src/common/comp/brute_force.cc

```diff
@@ -37,6 +37,30 @@ namespace knowhere {
 
 class BruteForceConfig : public BaseConfig {};
 
+namespace {
+
+template <typename T>
+expected<sparse::DocValueComputer<T>>
+GetDocValueComputer(const BruteForceConfig& cfg) {
+    if (IsMetricType(cfg.metric_type.value(), metric::IP)) {
+        return sparse::GetDocValueOriginalComputer<T>();
+    } else if (IsMetricType(cfg.metric_type.value(), metric::BM25)) {
+        if (!cfg.bm25_k1.has_value() || !cfg.bm25_b.has_value() || !cfg.bm25_avgdl.has_value()) {
+            return expected<sparse::DocValueComputer<T>>::Err(
+                Status::invalid_args, "bm25_k1, bm25_b, bm25_avgdl must be set when searching for bm25 metric");
+        }
+        auto k1 = cfg.bm25_k1.value();
```

Comment:
Please add constraints on possible value ranges for `k1`, `b` and `avgdl` as well. For example, `avgdl` should be non-zero, as far as I see from the code.
Also, there is a duplicate code down below that also needs such a check, is it possible to add something like 
`static bool CheckBM25Params(const BaseConfig& config) {...}` ?

Reply:
the value ranges are defined in `config.h` and automatically checked, for example:

```cpp
KNOWHERE_CONFIG_DECLARE_FIELD(bm25_k1)
    .allow_empty_without_default()
    .set_range(0.0, 3.0)
    .description("BM25 k1 to tune the term frequency scaling factor")
    .for_train_and_search()
    .for_iterator()
    .for_deserialize()
    .for_deserialize_from_file();
```